src/coreclr/vm/interpexec.cpp

```diff
@@ -6,6 +6,47 @@
 #include "threads.h"
 #include "gcenv.h"
 #include "interpexec.h"
+#include "callstubgenerator.h"
+
+void InvokeCompiledMethod(MethodDesc *pMD, int8_t *pArgs, int8_t *pRet)
+{
+    CONTRACTL
+    {
+        THROWS;
+        MODE_COOPERATIVE;
```

Comment:
If I use this for P/Invoke when suppressGcTransition is true for the target, will it be correct? Or do I need an alternative version of InvokeCompiledMethod which is pre-emptive for that scenario?

Reply:
No, there is nothing that requires that GC mode, this should actually be relaxed. I've just set the contract to what the caller in my change has. I'll change it.